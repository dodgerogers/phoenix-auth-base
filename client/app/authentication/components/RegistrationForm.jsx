import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { reduxForm } from 'redux-form/immutable';
import { Form, Message, Button, Icon } from 'semantic-ui-react';
import Input from '../../common/components/Input';
import { isRequired, isEmail, matchField, minLength } from '../../lib/utils/validation';
import { formIDs } from '../constants';

const passwordsMatch = matchField('password');
const minPasswordLength = minLength(8);

function RegistrationForm(props) {
  const { handleSubmit, error, submitting, dirty } = props;

  return (
    <div className="registration-form">
      <Form
        size='large'
        loading={submitting}
        onSubmit={handleSubmit}
      >
        {error && dirty && <Message negative>{error}</Message>}
        <Field
          component={Input}
          fluid={true}
          name="name"
          icon="user"
          iconPosition="left"
          placeholder="Display Name"
          validate={[isRequired]}
        />
        <Field
          component={Input}
          fluid={true}
          name="email"
          icon="mail"
          iconPosition="left"
          placeholder="Email"
          validate={[isRequired, isEmail]}
        />
        <Field
          component={Input}
          fluid={true}
          type="password"
          name="password"
          icon="lock"
          iconPosition="left"
          placeholder="Password"
          validate={[isRequired, minPasswordLength]}
        />
        <Field
          component={Input}
          fluid={true}
          type="password"
          name="passwordConfirmation"
          icon="lock"
          iconPosition="left"
          placeholder="Password Confirmation"
          validate={[isRequired, passwordsMatch]}
        />
        <Button
          fluid={true}
          color="teal"
          disabled={!props.valid}
        >
          Create account
        </Button>
      </Form>
    </div>
  );
}

RegistrationForm.propTypes = {
  handleSubmit: PropTypes.func,
  onSubmit: PropTypes.func,
  error: PropTypes.string,
  submitting: PropTypes.bool,
}

export { RegistrationForm as PureComponent };
export default reduxForm({ form: formIDs.REGISTRATION })(RegistrationForm);
