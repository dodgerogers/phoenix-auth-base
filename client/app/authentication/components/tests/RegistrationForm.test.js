import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PureComponent } from '../RegistrationForm';


describe('RegistrationForm', () => {
  const fullProps = {
    handleSubmit: jest.fn(),
    onSubmit: jest.fn(),
  };

  const pureComponent = (props) => shallow(
    <PureComponent {...props} />
  );

  it('matches component snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
