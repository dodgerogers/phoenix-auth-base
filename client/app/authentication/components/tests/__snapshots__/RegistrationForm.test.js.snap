// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm matches component snapshot 1`] = `
<div
  className="registration-form"
>
  <Form
    as="form"
    onSubmit={[Function]}
    size="large"
  >
    <Field
      component={[Function]}
      fluid={true}
      icon="user"
      iconPosition="left"
      name="name"
      placeholder="Display Name"
      validate={
        Array [
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="mail"
      iconPosition="left"
      name="email"
      placeholder="Email"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="lock"
      iconPosition="left"
      name="password"
      placeholder="Password"
      type="password"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="lock"
      iconPosition="left"
      name="passwordConfirmation"
      placeholder="Password Confirmation"
      type="password"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Button
      as="button"
      color="teal"
      disabled={true}
      fluid={true}
      role="button"
    >
      Create account
    </Button>
  </Form>
</div>
`;
